build.gradle

```diff
@@ -237,38 +237,40 @@ project(':cruise-control') {
       exclude group: 'log4j', module: 'log4j'
     }
-    compile project(':cruise-control-metrics-reporter')
-    compile project(':cruise-control-core')
-    compile "org.slf4j:slf4j-api:1.7.32"
-    compile "org.apache.logging.log4j:log4j-slf4j-impl:2.14.1"
-    compile "org.apache.zookeeper:zookeeper:${zookeeperVersion}"
-    compile "io.netty:netty-handler:${nettyVersion}"
-    compile "io.netty:netty-transport-native-epoll:${nettyVersion}"
-    compile "org.apache.kafka:kafka_$scalaBinaryVersion:$kafkaVersion"
-    compile "org.apache.kafka:kafka-clients:$kafkaVersion"
-    compile "org.scala-lang:scala-library:$scalaVersion"
-    compile 'org.apache.commons:commons-math3:3.6.1'
-    compile 'org.apache.httpcomponents:httpclient:4.5.13'
-    compile 'commons-codec:commons-codec:1.15'
-    compile 'com.google.code.gson:gson:2.8.8'
-    compile "org.eclipse.jetty:jetty-server:${jettyVersion}"
-    compile 'io.dropwizard.metrics:metrics-core:3.2.6'
-    compile 'com.nimbusds:nimbus-jose-jwt:9.13'
-    compile 'com.101tec:zkclient:0.11'
-    compile 'io.swagger.parser.v3:swagger-parser-v3:2.0.27'
-    compile 'io.github.classgraph:classgraph:4.8.115'
-
-    testCompile project(path: ':cruise-control-metrics-reporter', configuration: 'testOutput')
-    testCompile project(path: ':cruise-control-core', configuration: 'testOutput')
-    testCompile "org.scala-lang:scala-library:$scalaVersion"
-    testCompile 'junit:junit:4.13.2'
-    testCompile 'org.easymock:easymock:4.3'
-    testCompile "org.apache.kafka:kafka_$scalaBinaryVersion:$kafkaVersion:test"
-    testCompile "org.apache.kafka:kafka-clients:$kafkaVersion:test"
-    testCompile 'commons-io:commons-io:2.11.0'
-    testCompile 'org.apache.httpcomponents:httpclient:4.5.13:tests'
-    testCompile 'org.bouncycastle:bcpkix-jdk15on:1.69'
-    testCompile 'org.apache.kerby:kerb-simplekdc:2.0.1'
+    implementation project(':cruise-control-metrics-reporter')
+    implementation project(':cruise-control-core')
+    implementation "org.slf4j:slf4j-api:1.7.32"
+    implementation "org.apache.logging.log4j:log4j-slf4j-impl:2.14.1"
+    implementation "org.apache.zookeeper:zookeeper:${zookeeperVersion}"
+    implementation "io.netty:netty-handler:${nettyVersion}"
+    implementation "io.netty:netty-transport-native-epoll:${nettyVersion}"
+    implementation "org.apache.kafka:kafka_$scalaBinaryVersion:$kafkaVersion"
+    implementation "org.apache.kafka:kafka-clients:$kafkaVersion"
+    implementation "org.scala-lang:scala-library:$scalaVersion"
+    implementation 'org.apache.commons:commons-math3:3.6.1'
+    implementation 'org.apache.httpcomponents:httpclient:4.5.13'
+    implementation 'commons-codec:commons-codec:1.15'
+    implementation 'com.google.code.gson:gson:2.8.8'
+    implementation "org.eclipse.jetty:jetty-server:${jettyVersion}"
+    implementation "org.eclipse.jetty:jetty-servlet:${jettyVersion}"
+    implementation "org.eclipse.jetty:jetty-security:${jettyVersion}"
```

Comment:
It is interesting that we lost these dependencies after a Gradle version bump.

Do we actually need to add `implementation "org.eclipse.jetty:jetty-security:${jettyVersion}"` as `jetty-security` is already pulled by `jetty-servlet`?

Reply:
> It is interesting that we lost these dependencies after a Gradle version bump.

Good point, debugged further and replaced `compile` with `api` instead of `implementation` to inherit depedencies from core and metrics sub-projects

Reply:
Thanks for making the migration using the the equivalent of the deprecated `compile` (i.e. `api`)! This is definitely cleaner for the migration purposes.

I read a little bit about the distinction between `implementation` and `api`. Maybe in the future we can revisit this part to explore the feasibility of using `implementation`, which claims to provide faster builds and prevent leaking the internal implementation.

Reply:
I agree - I did the same and while for the cruise-control-core and cruise-control-metrics-reporter it does makes sense to define them as `api` dependencies as they are more like libraries for the main project - it;s worth discussing if other should be defined using api or implementation.
There may be integration out there that use cruise-control as a library dep and in that case it makes sense to keep the transitive libraries